Week04/MoviesApp/app/src/main/java/tech/danielwaiguru/moviesapp/MovieApp.kt

```diff
@@ -15,5 +25,34 @@ class MovieApp: Application() {
     override fun onCreate() {
         super.onCreate()
         instance = this
+        delayedInit()
+    }
+
+    /**
+     * Setting recurring work to refresh movies data every one hour
+     */
+    private fun setRecurringWork(){
+        val constraints = androidx.work.Constraints.Builder()
+            .setRequiresBatteryNotLow(true)
+            .setRequiredNetworkType(NetworkType.CONNECTED)
+            .build()
+        val syncMoviesWorker =
+            PeriodicWorkRequestBuilder<SyncMoviesWorker>(20, TimeUnit.MINUTES)
+                .setConstraints(constraints)
+                .build()
+        val workManager = WorkManager.getInstance(this)
+        workManager.enqueueUniquePeriodicWork(
+            SyncMoviesWorker.WORK_NAME,
+            ExistingPeriodicWorkPolicy.KEEP,
+            syncMoviesWorker
+        )
+    }
+    /**
+     * method to start a coroutine
+     */
+    private fun delayedInit(){
```

Comment:
Not sure why you need to create a coroutine, or why this function is called delayedInit()